### DISCUSSION TITLE
How can I get a ip address from INADDR_ANY ?

### DISCUSSION THREAD

**Discussion by Meraroo** [ANLAB-KAIST/KENSv3#31] [Project Q&A 1. Application Layer (Socket)]
![image](https://user-images.githubusercontent.com/81629557/159122824-2c45bb3c-a870-4dc7-9a73-f15d9b965c6e.png)
![image](https://user-images.githubusercontent.com/81629557/159122809-d0033049-88e8-4444-b70a-8c101eae0657.png)

I think the client's IP is bound by INADDR_ANY.
Is there any way to get a real client's ip address?
(I tried line 68 with getsockname (client, &caddr, &clen) but output is same..)

**Comment by Pusnow** [comment]
*I guessed your **server**'s IP address is bound by `INADDR_ANY`.*

To obtain client's IP address, you should use `getpeername`.

**Reply by Meraroo** [reply]
I found when I use inet_ntoa () function, then accept () returns -1 so serverMain () returns -1... I totally don't know why such things happen........
